src/bin/pg_waldump/pg_waldump.c

```diff
@@ -1388,6 +1423,27 @@ main(int argc, char **argv)
 	if (!xlogreader_state)
 		pg_fatal("out of memory while allocating a WAL reading processor");
 
+	if (save_records_file)
+	{
+		/*
+		 * NEON: We dump records in the format recognized by walredo process.
+		 * It is libpq compatible format: one character tag + 4 bytes length.
```

Comment:
```suggestion
		 * one character tag + 4 bytes length.
```
It's very similar to libpq message format, but that was accidental, there's no expectation that these messages would ever be sent as libpq messages.

I'm also not sure if the length is accounted the same way; I don't remember if the length includes the length itself or the tag character in either format.